openedx/core/djangoapps/content/course_structures/tasks.py

```diff
@@ -60,7 +61,20 @@ def _generate_course_structure(course_key):
         }
 
 
-@task(name=u'openedx.core.djangoapps.content.course_structures.tasks.update_course_structure')
+def ensure_lms_queue():
+    """
+    Ensure the worker associated with the chosen queue is loaded with lms params, if applicable.
+    """
+    queues = getattr(settings, 'CELERY_QUEUES', None)
+    print queues
+    lms_queue = next((queue for queue in queues if '.lms.' in queue), None)
+    return lms_queue
+
+
+@task(
+    name=u'openedx.core.djangoapps.content.course_structures.tasks.update_course_structure',
+    queue=ensure_lms_queue(),
```

Comment:
Why choose to have the code itself specify the queue, rather than use celery's router settings technique?  The latter would allow us to configure centrally where each task should be enqueued by task-type.